phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -502,9 +503,10 @@ private void populateRowsToLock(MiniBatchOperationInProgress<Mutation> miniBatch
           Mutation m = miniBatchOp.getOperation(i);
           if (this.builder.isAtomicOp(m) || this.builder.isEnabled(m)) {
               ImmutableBytesPtr row = new ImmutableBytesPtr(m.getRow());
-              if (!context.rowsToLock.contains(row)) {
+//              This seems to be unnecessary as we are using a set here
+//              if (!context.rowsToLock.contains(row)) {
```

Comment:
Thanks for improving this - can you remove this commented code, as don't think this needs to be checked in.

Reply:
Thanks @mnpoonia . Looks good. Please remove the commented as suggested by @gourabtaparia

Reply:
Done